arbos/chains.go

```diff
@@ -19,11 +28,31 @@ func getStaticChainConfig(chainId *big.Int) (*params.ChainConfig, error) {
 	return nil, fmt.Errorf("unsupported L2 chain ID %v", chainId)
 }
 
-func GetChainConfig(chainId *big.Int, genesisBlockNum uint64) (*params.ChainConfig, error) {
-	staticChainConfig, err := getStaticChainConfig(chainId)
-	if err != nil {
-		return nil, err
+func GetChainConfig(chainId *big.Int, genesisBlockNum uint64, l2ChainInfoFiles []string) (*params.ChainConfig, error) {
+	for _, l2ChainInfoFile := range l2ChainInfoFiles {
+		chainsInfoBytes, err := os.ReadFile(l2ChainInfoFile)
+		if err != nil {
+			return nil, err
+		}
+		var chainsInfo map[uint64]ChainInfo
+		err = encoding_json.Unmarshal(chainsInfoBytes, &chainsInfo)
+		if err != nil {
+			return nil, err
+		}
+		if _, ok := chainsInfo[chainId.Uint64()]; !ok {
+			continue
+		}
+		chainConfig := chainsInfo[chainId.Uint64()].ChainConfig
+		chainConfig.ArbitrumChainParams.GenesisBlockNum = genesisBlockNum
```

Comment:
Let's do this in a separate PR, but we should be able to replace the dynamic genesis block number system with just putting the genesis block number in the chain config, because we've already done the classic chain imports.